### DISCUSSION TITLE
Live Setting

### DISCUSSION THREAD

**Discussion by tipodanet** [k-ivan/visual-bookmarks-chrome#189] [Ideas]
С приходом современных браузеров появилась возможность сразу видеть результат изменения настроек. Как это технически реализовано -- мне не известно. 
Но настройки не имеют кнопки "ОК" или "Готово" или "Применить".
Я поддерживаю, Намного быстрее и макс. удобно. 
Юзер видит результат сразу. 

В этом режиме искать баги -- просто THE BEST. В расширении с полсотни настроек -- какая из них глючит , вычислить сложно.
Жду не дождусь.

**Comment by k-ivan** [comment]
Закрываю сразу, так как это очень тяжко сделать в текущей кодовой базе. И даже  не планирую в будущем. Слишком много сайд эффектов, которые предстоит контролировать.

**Comment by tipodanet** [comment]
а может там просто обновление после каждой настройки. Как бы кнопка "Готово"  автоматом после каждого изменения . С учетом быстрых многоядерных процессоров и пустой памяти 32-128 ГБ , юзер и не заметит, что идет перезагрузка всего расширения.
Лично у меня ВЕСЬ браузер в оперативной памяти. Включая его установку. Виртуальные диски -- сейчас модно.

**Comment by k-ivan** [comment]
Кстати говоря, кнопку - Готово жать не обязательно, она по факту ничего не сохраняет, а просто ведет на страницу расширения
Настройки сохраняются сразу, по мере изменения ползунков.

Можно рассмотреть вариант с хард рефрешом страницы закладок в принципе. Правда если их несколько по открывать, все обновляться будут.

Попробую на досуге, посмотрим, что из этого выйдет.

**Comment by tipodanet** [comment]
хард рефреш -- звучит , так как и звучит. То что надо. 
Если открыть 2 вкладки  TILE (все браузеры это умеют).
То  вкладки независимы, 1 не видит настроек из 2,  даже после "Готово"

**Reply by tipodanet** [reply]
А как такой вариант : разделить окно горизонтально на 2 части. 
Сверху 1 ряд с картинками. Снизу настройки. 
У меня на ПК  экран FullHD. 
А на ноуте 2860*1800. Т.е. FullHD занимает по пикселям чуть больше 1/4 . Не говоря уже про 4К-мониторы.

**Reply by k-ivan** [reply]
точно нет

**Reply by tipodanet** [reply]
Я вспоминаю этот чертовый Вивальди, когда его представитель по PR, с зарплатой в 5000 Э, выкладывает свои скрины , я вижу монитор около 900*1600  . Именно такая настройка у этого браузера для PIP. Т.е. внешнее видео не растягивается шире 900*1600 . 
Им уже  весь форум лет 6 пишет , снимите ограничение, им пофиг .
Пример , когда все на старых ноутах, сами старые, слепые и в очках, делают современный браузер  --- "а вот в наши 90-е я начал программить ....".

**Comment by k-ivan** [comment]
если сделать рефреш при изменении настроек, все вкладки перезагрузятся, если это реализовать, ни факт, что частый кейс, но на слабых машинах может если много закладок и открыть несколько вкладок, может быть не приятно. Но это конечно высосано из пальца. Посмотрим, попробую, как будет возможность.

**Reply by tipodanet** [reply]
> , все вкладки перезагрузятся,

а что это значит ? У меня бывает открыты  несколько ВЗ, но это случайно, по недосмотру. Смысла в копиях ВЗ нет. 
Если включить "открывать новую вкладку здесь", то  копий не будет.

**Comment by tipodanet** [comment]
Я понял как  -- аналог OPERA  .  У них настройки двойные. Первые узкие справа.  Вторые --  основные.
Настройки можно в виде  накладных -- не меняя масштаб картинок. Правую часть можно перекрыть процентов на 25%.  
И конечно, изменение окна мышью . Юзер сам растянет, сколько ему надо настроек ---  25% или 70%.
LiveSetting сейчас у всех браузеров, без исключения. Надо соответствовать .
Боковые настройки только у  OPERA. Выглядит удобно.

У  вас настройки из 3 частей . JQUERY имеет такую хрень, как Табы-вкладки :) . Они узкие

**Reply by k-ivan** [reply]
мне эта тема не интересна, это огромная куча работы, ради легкого ничего. Контролировать тонны сайд эффектов, я не хочу. 

Настройки могут открываться внутри самой страницы уже сейчас, правда это будет окошко браузера по середине экрана(и контролировать его особо не получится). Я, пока выбираю текущий вариант вместо окошка браузера по середине.

**Comment by tipodanet** [comment]
Это называется ориtнтированность на пользователя. Как там , Юзабилити.
Особенно вкладки "Вид".
Любой юзер поиграет с числом квадратов и шириной в Живом исполнении, И выберет свой вариант.

Очевидно, что Число колонок должно быть Слайдером, а не списком.

**Comment by tipodanet** [comment]
а почему так сложно ? Все SLIDERS от старого  JQUERY  обновляются в  real-time. Как и должно быть

**Reply by k-ivan** [reply]
Вы не много не понимайте контекста, поэтому поверьте на слово. Jquery тут вообще не при чем. Это реально, но я не вижу смысла, так все переделывать, ради незначительного преимущества.
